src/lib/api/prefetch.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2018 by Filestack.
+ * Some rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ClientOptions, Session } from '../client';
+import { PickerOptions } from './../picker';
+import { requestWithSource } from '../api/request';
+import * as cloneDeep from 'lodash.clonedeep';
+
+export type PrefetchOptionsSetting = {
+  inapp_browser?: boolean;
+  customsource?: boolean;
+};
+
+export type PrefetchOptionsPermissions = {
+  intelligent_ingestion?: boolean;
+  blocked?: boolean;
+  blacklisted?: boolean;
+  whitelabel?: boolean;
+  transforms_ui?: boolean;
+};
+
+enum PrefetchOptionsEvents {
+  PICKER = 'picker',
+  TRANSFORM_UI = 'transform_ui',
+}
+
+type PrefetchOptions = {
+  pickerOptions?: PickerOptions;
+  settings?: PrefetchOptionsSetting;
+  permissions?: PrefetchOptionsPermissions;
+  events?: PrefetchOptionsEvents[];
+};
+
+interface SendObject {
```

Comment:
Consider to change name ie to 
PrefetchRequest so if u will use that type in other file, name will point to what its represent. The same with responseObject ie to PrefetchResponse

Reply:
done